### DISCUSSION TITLE
"Addon app support" what does it mean?

### DISCUSSION THREAD

**Discussion by dimakaob** [BimmerGestalt/AAIdrive#482] [Q&A] [answered]
Hello! 
I am sorry if my question is stupid. 
Just found out, that there is the latest 1.3.1 version of AAidrive. 
So what does "Addon app support" mean? 
Thank you.

**Comment by hufman** [comment, marked as answer]
AAIdrive can now provide easy access for other phone apps to read car data, and can also launch more complex apps to start up and connect to the car directly. There are a few "official" [addon apps](https://github.com/BimmerGestalt/IDriveConnectAddons), meant mostly as examples with the goal that other people might be inspired to make their own apps, but the screen mirroring addon seems especially popular by itself. There's a link from within AAIdrive's Addons tab to that page, for convenience.

**Comment by dimakaob** [comment]
Wow!
That sounds so cool!
Definitely will give it a try.

But how to install those addons?
Screen_mirror, for example. I see 3 files and one folder src..

Update. 
Found an .apk file =) 
Will see how it wors later. 

Thank you so much! 

вт, 28 дек. 2021 г. в 18:42, Walter Huf ***@***.***>:

**Reply by hufman** [reply]
Indeed, those files are the source code, for people to learn how the apps are built and how to build their own.
You would install the APKs from [the Releases page](https://github.com/BimmerGestalt/IDriveConnectAddons/releases).

**Comment by dimakaob** [comment]
Yes, everything works! But is there any solution with connection lags? I
missed two turns because of it  =((
I mean that the picture on the car screen updates not the exact time as on
smartphone.

чт, 30 дек. 2021 г., 18:32 Walter Huf ***@***.***>:

**Reply by hufman** [reply]
There isn't much of a solution for that, I'm afraid, it is sending a jpg screenshot of the phone over Bluetooth which is very slow. Additionally, to ensure that people aren't using this to watch movies while driving, it only updates every 2 seconds while in motion, which I figure should be fine for navigation purposes. It can detect if Android Auto is running and will continue updating the screen as fast as possible, which still isn't very good.

**Comment by dimakaob** [comment]
Thank you for your reply.
So this 2 sec update can't be changed to 1 sec update? Right?

пн, 3 янв. 2022 г., 18:13 Walter Huf ***@***.***>:

**Reply by hufman** [reply]
Because the code is open source, you can change it and recompile the app with any changes you want :) But I plan to keep the official version on the safe side, especially to keep BMW happy.

**Comment by dimakaob** [comment]
I understand %)
Thank you for all you do! I really mean it.


вт, 4 янв. 2022 г., 19:15 Walter Huf ***@***.***>:
